File: book/03_variables/src/main.go
package main

import "fmt"

const THREE_HOURS_IN_SECONDS = 60 * 60 * 3

func main() {
	// mutable by default
	x := 5

	fmt.Printf("The value of x is: %d\n", x)

	x = 6

	{
		x := x * 2
		fmt.Printf("The value of x in the inner scope is: %d\n", x)
	}

	fmt.Printf("The pointer value of x is: %v\n", &x)

	/* fmt.Println(&x)
	fmt.Println(&x)
	fmt.Println(*&x)
	*/

	fmt.Println(THREE_HOURS_IN_SECONDS)
}

File: book/03_variables/src/main.rs
const THREE_HOURS_IN_SECONDS: u32 = 60 * 60 * 3;

fn main() {
    let mut x = 5;

    println!("The value of x is: {}", x);

    x = 6;

    {
        let x = x * 2;
        println!("The value of x in the inner scope is: {}", x);
    }

    println!("The pointer value of x is: {:p}", &x);

    println!("{}", THREE_HOURS_IN_SECONDS);
}
